ding/framework/task.py

```diff
@@ -225,8 +225,12 @@ def __exit__(self, exc_type, exc_val, exc_tb):
         self.stop()
 
     def stop(self) -> None:
+        self.emit("exit")
         if self._thread_pool:
             self._thread_pool.shutdown()
+        self.middleware.clear()
```

Comment:
add some comments for the sequence of calling clear method

Reply:
Done